### PR TITLE
 Replace Quill RichTextInput With TipTap RichTextInput

### PR DESCRIPTION
**Pull request by djhi**
## The problem

Our old `RichTextInput` was based on Quill but:
- it wasn't accessible (button without labels, etc.)
- it wasn't translatable (labels in Quill are in the CSS)
- it wasn't using MUI components for its UI so it looks off

## The solution

The new `RichTextInput` uses [TipTap](https://github.com/ueberdosis/tiptap), a UI less library to build rich text editors. It gives us the freedom to implement the UI how we want with MUI components. That solves all the above issues

## Tasks

- [x] Implementation
- [x] Tests
- [x] Upgrade guide
- [x] Documentation

### CODE REVIEW

**Comment by fzaninotto** [comment]
needs rebase

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
         "run-graphql-demo-watch": "concurrently \"yarn run watch\" \"yarn run run-graphql-demo\"",
         "run-crm": "cd examples/crm && yarn start",
         "build-crm": "cd examples/crm && yarn build",
-        "storybook": "start-storybook -p 9009",
+        "storybook": "start-storybook -p 9010",
```
Comment (fzaninotto) [inline package.json:25]: why?

File: packages/ra-input-rich-text/README.md [hunk 2]
```diff
@@ -1,79 +1,141 @@
-# `<RichTextInput>` for react-admin
+# ra-richtext-tiptap
```
Comment (fzaninotto) [inline packages/ra-input-rich-text/README.md:1]: that's not the name of the package

File: packages/ra-input-rich-text/README.md [hunk 3]
```diff
@@ -1,79 +1,141 @@
...
-For editing HTML with [react-admin](https://github.com/marmelab/react-admin), use the `<RichTextInput>` component. It embarks [quill](https://quilljs.com/), a popular cross-platform Rich Text Editor.
-
-![`<RichTextInput>` example](https://marmelab.com/react-admin/img/rich-text-input.png)
+A rich text editor for [React Admin](http://marmelab.com/react-admin), based on [TipTap](https://www.tiptap.dev/)
 
 ## Installation
 
 ```sh
-npm install ra-input-rich-text --save-dev
+npm install ra-richtext-tiptap
```
Comment (fzaninotto) [inline packages/ra-input-rich-text/README.md:8]: bad package name

File: packages/ra-input-rich-text/README.md [hunk 4]
```diff
@@ -1,79 +1,141 @@
...
-    EditButton,
-    TextInput,
-} from 'react-admin';
-import RichTextInput from 'ra-input-rich-text';
-
-const PostTitle = ({ record }) => {
-    return <span>Post {record ? `"${record.title}"` : ''}</span>;
-};
+import { Edit, SimpleForm, TextInput } from 'react-admin';
+import { RichTextInput } from 'ra-richtext-tiptap';
```
Comment (fzaninotto) [inline packages/ra-input-rich-text/README.md:19]: bad import

File: packages/ra-input-rich-text/README.md [hunk 5]
```diff
@@ -1,79 +1,141 @@
...
 
-This library is licensed under the MIT License, and sponsored by [marmelab](https://marmelab.com).
+export const MyEditorOptions = {
+	...DefaultEditorOptions,
+	extensions: [
+		...DefaultEditorOptions.extensions,
+        HorizontalRule,
+	],
+};
+```
```
Comment (fzaninotto) [inline packages/ra-input-rich-text/README.md:141]: missing license section

File: packages/ra-input-rich-text/package.json [hunk 6]
```diff
@@ -2,62 +2,52 @@
...
     "devDependencies": {
+        "@mui/icons-material": "^5.0.1",
         "@mui/material": "^5.0.2",
         "@testing-library/react": "^11.2.3",
-        "@types/prop-types": "^15.6.0",
-        "npm-dts": "^1.3.10",
-        "ra-core": "^4.0.0-alpha.1",
-        "ra-ui-materialui": "^4.0.0-alpha.1",
-        "rimraf": "^2.6.3",
+        "data-generator-retail": "^4.0.0-alpha.0",
```
Comment (fzaninotto) [inline packages/ra-input-rich-text/package.json:44]: please update to alpha.2

File: packages/ra-ui-materialui/src/defaultTheme.ts [hunk 7]
```diff
@@ -45,6 +45,22 @@ export const defaultTheme = {
                         opacity: 0.3,
                         borderRadius: 'inherit',
                     },
+                    '&:focus::after': {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/defaultTheme.ts:48]: Can you elaborate why you need that?

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (djhi) [reply package.json:25]: Had a known issue with storybook only returning 404. Changing the port solved it

File: packages/ra-ui-materialui/src/defaultTheme.ts (hunk 7 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/defaultTheme.ts:48]: Added more comments

File: packages/ra-input-rich-text/README.md (hunk 5 above)
Comment (djhi) [reply packages/ra-input-rich-text/README.md:141]: Added. We have several packages without this section btw

**Review by WiXSL — COMMENTED** [review]
There is a library dependency related to Quill in `test-setup.js`
```js
/**
 * As jsDom do not support mutationobserver and
 * quill requires mutationobserver, thus a shim is needed
 */
require('mutationobserver-shim');

File: cypress/integration/edit.js [hunk 8]
```diff
@@ -150,7 +150,7 @@ describe('Edit Page', () => {
         cy.get(CreatePostPage.elements.input('body', 'rich-text-input')).should(
             el =>
                 // When the Quill editor is empty, it add the "ql-blank" CSS class
```
Comment (WiXSL) [inline cypress/integration/edit.js:152]:
```suggestion
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: cypress/integration/create.js [hunk 9]
```diff
@@ -338,9 +340,9 @@ describe('Create Page', () => {
         // Quill take a little time to boot and Cypress is too fast which can leads to unstable tests
```
Comment (WiXSL) [inline cypress/integration/create.js:340]: Quill reference spotted

**Review by WiXSL — CHANGES_REQUESTED** [review]

**Review by fzaninotto — APPROVED** [review]

**Review by WiXSL — APPROVED** [review]

**Pull request merged**